### PR TITLE
core: fix potential goroutine leak

### PR DESCRIPTION
**Pull request by qinglin89**
### Description

fix potential goroutine leak.
When `interruptCh` was closed before all transactions be executed, `txChan <- i` would be blocked forever since there would be no process consume it. Make sure `prefetch` routine would eventually exit by `select` `interrupt` channel

### Rationale

N/A

### Example

N/A

### Changes

Notable changes:

### CODE REVIEW

**Comment by guagualvcha** [comment]
I think the description of this PR is not clear.

File: core/state_prefetcher.go [hunk 1]
```diff
@@ -55,6 +55,7 @@ func (p *statePrefetcher) Prefetch(block *types.Block, statedb *state.StateDB, c
 	)
 	transactions := block.Transactions()
 	txChan := make(chan int, prefetchThread)
+	defer close(txChan)
```
[the remark this answers is not in the document]
Comment (qinglin89) [reply core/state_prefetcher.go:58]: `close` is used for `for` loop on line:68 to return

[the remark this answers is not in the document]
Comment (brilliant-lx) [reply core/state_prefetcher.go:58]: what if the prefetchThread read the `txChan` after it is closed?

**Comment by qinglin89** [comment]
> A UT is highly suggested to demonstrate the modification works.

Added

**Review by brilliant-lx — APPROVED** [review]
LGTM

File: go.mod [hunk 2]
```diff
@@ -35,6 +35,7 @@ require (
 	github.com/golang/protobuf v1.4.3
 	github.com/golang/snappy v0.0.4
 	github.com/google/gofuzz v1.1.1-0.20200604201612-c04b05f3adfa
+	github.com/google/pprof v0.0.0-20220829040838-70bd9ae97f40
```
Comment (guagualvcha) [inline go.mod:38]: Can we remove the change of `go.mod` and `go.sum`, seems this PR is not related with dependency.

Comment (guagualvcha) [reply go.mod:38]: Sorry. I missed  that you used google profile, it is ok with me then.

**Review by guagualvcha — APPROVED** [review]
